File: json/rebol/create/json_create.r
#! /usr/local/bin/rebol 
;
;	json_create.r
;
;				Sep/25/2011
; ------------------------------------------------------------------
REBOL [
Title:  "Json Create"
Author: "Uchida Masatomo"
]

; ------------------------------------------------------------------
tag_gen_proc: func [tag value][
	rejoin ["^"" tag "^": ^"" value "^"" ]
]
;
; ------------------------------------------------------------------
gen_string: func [id name population date_mod delimit][
;
	bb: tag_gen_proc "name" name
	cc: tag_gen_proc "population" population
	dd: tag_gen_proc "date_mod" date_mod
	rejoin ["^"" id  "^": {" bb "," cc "," dd "}" delimit]
;
]
; ------------------------------------------------------------------
print "*** 開始 ***"
;
file_out: %/var/tmp/json/cities.json
;
write/lines file_out "{"
write/append/lines file_out gen_string "t0921" "宇都宮" "42379" "1966-10-5" ","
write/append/lines file_out gen_string "t0922" "小山" "53723" "1966-5-18" ","
write/append/lines file_out gen_string "t0923" "佐野" "37814" "1966-9-22" ","
write/append/lines file_out gen_string "t0924" "足利" "61842" "1966-9-9" ","
write/append/lines file_out gen_string "t0925" "日光" "42351" "1966-8-4" ","
write/append/lines file_out gen_string "t0926" "下野" "35187" "1966-1-21" ","
write/append/lines file_out gen_string "t0927" "さくら" "81256" "1966-7-23" ","
write/append/lines file_out gen_string "t0928" "矢板" "24784" "1966-10-26" ","
write/append/lines file_out gen_string "t0929" "真岡" "75823" "1966-12-15" ","
write/append/lines file_out gen_string "t0930" "栃木" "23412" "1966-11-21" ","
write/append/lines file_out gen_string "t0931" "大田原" "17854" "1966-8-9" ","
write/append/lines file_out gen_string "t0932" "鹿沼" "56792" "1966-5-4" ","
write/append/lines file_out gen_string "t0933" "那須塩原" "39247" "1966-7-12" ","
write/append/lines file_out gen_string "t0934" "那須烏山" "82531" "1966-9-21" "}"

print "*** 終了 ***"
quit

; ------------------------------------------------------------------

File: json/swipl/create/json_create.pl
/* ------------------------------------------------------------------- */
/*
	json_create.pl

						Feb/28/2020
*/
/* ------------------------------------------------------------------- */
:- use_module([ library(http/json) ]).
/* ------------------------------------------------------------------- */
data_out_proc(File_out,Dict) :-
	tell(File_out),
	json_write_dict(current_output, Dict),
	told.
/* ------------------------------------------------------------------- */
data_prepare_proc(Dict) :- Dict = _{
	t0921:_{name:"宇都宮", population : 15876, date_mod : '1942-5-16' },
    	t0922:_{name:"小山", population : 81274, date_mod : '1942-7-12' },
    	t0923:_{name:"佐野", population : 71658, date_mod : '1942-9-22' },
    	t0924:_{name:"足利", population : 41359, date_mod : '1942-4-25' },
    	t0925:_{name:"日光", population : 61937, date_mod : '1942-3-28' },
    	t0926:_{name:"下野", population : 91854, date_mod : '1942-1-7' },
    	t0927:_{name:"さくら", population : 21657, date_mod : '1942-2-19' },
    	t0928:_{name:"矢板", population : 51648, date_mod : '1942-5-30' },
    	t0929:_{name:"真岡", population : 18647, date_mod : '1942-6-3' },
    	t0930:_{name:"栃木", population : 96425, date_mod : '1942-7-11' },
    	t0931:_{name:"大田原", population : 32641, date_mod : '1942-8-18' },
    	t0932:_{name:"鹿沼", population : 61342, date_mod : '1942-9-30' },
    	t0933:_{name:"那須塩原", population : 68745, date_mod : '1942-10-20' },
    	t0934:_{name:"那須烏山", population : 61932, date_mod : '1942-12-8' }
	}.

/* ------------------------------------------------------------------- */
:- initialization main.

main(Argv) :-
	format('*** 開始 ***\n'),
	format('Argv: ~w\n', [Argv]),
	[File_out|_] = Argv,
	write(File_out),nl,

	data_prepare_proc(Dict),

	data_out_proc(File_out,Dict),

	format('*** 終了 ***\n'),
	halt.

/* ------------------------------------------------------------------- */
